bot/exts/evergreen/youtube.py

```diff
@@ -0,0 +1,77 @@
+from typing import Dict, List, Optional
+
+from discord import Embed
+from discord.ext.commands import Bot, Cog, Context, command, cooldown
+from discord.utils import escape_markdown
+
+from bot.constants import Colours, Tokens
+
+KEY = Tokens.youtube
+SEARCH_API = (
+    "https://youtube.googleapis.com/youtube/v3/search?"
+    "part=snippet&type=video&maxResults=5&q={search_term}&key={key}"
+)
+YOUTUBE_URL = "https://www.youtube.com/watch?v={id}"
+RESULT = "`{index}` [{title}]({url}) - {author}"
+
+
+class YouTubeSearch(Cog):
+    """Sends the top 5 results of a query from YouTube."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.http_session = bot.http_session
+
+    async def search_youtube(self, search_term: str) -> Optional[List[Dict[str, str]]]:
+        """Queries API for top 5 results matching the search term."""
+        results = []
+        async with self.http_session.get(
+            SEARCH_API.format(search_term=search_term, key=KEY)
+        ) as response:
+            data = await response.json()
```

Comment:
Do we need to deal with the possibility of a non-200 response here?

Reply:
It seemed like it never occurred when I was testing it, no matter how random the query, though I can make a catchall.

Reply:
I'll implement one for no results from the query and the rest as a catchall "Something went wrong here..."